docs: Update how to work on guide articles (topics) on forum (#36735)
freeCodeCamp/freeCodeCamp: docs/style-guide-for-curriculum-challenges.md

Creating these challenges requires immense creativity and attention to detail. But you'll have plenty of help. You have support from a whole team of contributors, whom you can bounce ideas off of and demo your challenges to. Stay active in the [contributors room](https://gitter.im/freecodecamp/contributors) and ask lots of questions.

With your help, we can design an interactive coding curriculum that will help millions of people learn to code for years to come.

## Reply Templates for Moderators Reviewing Pull Requests

> > You can make your own with GitHub's built in [**Saved replies**](https://github.com/settings/replies/) feature or use the ones below.

Reviewer: ```suggestion
> You can make your own with GitHub's built-in [**Saved replies**](https://github.com/settings/replies/) feature or use the ones below.
```

### Thank you

```markdown
Thank you for your contribution to the page! 👍

> We're happy to accept these changes, and look forward to future contributions. 🎉

Reviewer: ```suggestion
We're happy to accept these changes and look forward to future contributions. 🎉
```

```

### Thank you and congrats

> For thanking and encouraging first-time contributors.

```markdown
Hi @username. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍

> We're happy to accept these changes, and look forward to future contributions. 📝

Reviewer: ```suggestion
We're happy to accept these changes and look forward to future contributions. 📝
```

```

### Build Error

```markdown
Hey @username

So I'd love to be able to merge your changes but it looks like there is an error with the Travis CI build. ⚠️

Once you resolve these issues, I will be able to review your PR and merge it. 😊

---

> Feel free to reference the [Style guide for writing articles](https://github.com/freeCodeCamp/freeCodeCamp#article-title) for this repo on formatting an article correctly so your Travis CI build passes. ✅
>
> Also, it's good practice on GitHub to write a brief description of your changes when creating a PR. 📝
```

### Syncing Fork

> > When PR is not up to date with `master` branch.

Reviewer: ```suggestion
> When PR is not up to date with the `master` branch.
```

``````markdown
Hey @username

So I'd love to be able to merge your changes but it looks like there is an error with the Travis CI build. ⚠️

```bash
Error: ENOTDIR: not a directory, open 'src/pages/java/data-abstraction/index.md'
```

> This particular error was not actually caused by your file but was an old error caused by merging faulty code to the `master` branch. It has since been resolved.

Reviewer: ```suggestion
This particular error was not caused by your file but was an old error caused by merging faulty code to the `master` branch. It has since been resolved.
```

git pull upstream master
```

If you're using a GUI, you can simply `Add a new remote...` and use the link `git://github.com/freeCodeCamp/freeCodeCamp.git` from above.

Once you sync your fork and pass the build, I will be able to review your PR and merge it. 😊

---

> Feel free to reference the [Syncing a Fork](https://help.github.com/articles/syncing-a-fork/) article on GitHub for more insight on how to keep your fork up-to-date with the upstream repository. 🔄
>
> Also, it's good practice on GitHub to write a brief description of your changes when creating a PR. 📝
``````

### Merge Conflicts

> When PR has merge conflicts that need to be resolved.¹

```markdown
Hey @username

So I'd love to be able to merge your changes but it looks like you have some merge conflicts. ⚠️

Once you resolve these conflicts, I will be able to review your PR and merge it. 😊

---

> If you're not familiar with the merge conflict process, feel free to look over GitHub's guide on ["Resolving a merge conflict"](https://help.github.com/articles/resolving-a-merge-conflict-on-github/). 🔍️
>
> Also, it's good practice on GitHub to write a brief description of your changes when creating a PR. 📝
```

> ¹ If a first-time-contributor has a merge conflict maintainers will resolve the conflict for them.

Reviewer: ```suggestion
¹ If a first-time-contributor has a merge conflict, maintainers will resolve the conflict for them.
```

### Duplicate

> When PR is repetitive or a duplicate.

```markdown
Hey @username

It seems that similar changes have already been accepted earlier for this article you're editing, sorry about that. 😓

If you feel you have more to add, please feel free to open up a new PR.

Thanks again! 😊

---

> If you have any questions, feel free to reach out through [Gitter](https://gitter.im/FreeCodeCamp/Contributors) or by commenting below. 💬
```

### Closing invalid pull requests

> When PR is invalid.

```markdown
Hey @username

> You haven't actually added any content so I will be  invalid pull requests this PR and marking it as `invalid`. 😓️

Reviewer: ```suggestion
You haven't actually any content so I will be closing this PR and marking it as `invalid`. 😓️
```
